### PR TITLE
numactl: Add versions 2.0.16-2.0.18

### PR DESCRIPTION
**Pull request by dc-snl**
Latest version as of today is 2.0.18. Version 2.0.17 contains a fix for building with musl libc, mentioned https://github.com/numactl/numactl/issues/185.

Skipping [2.0.15](https://github.com/numactl/numactl/releases/tag/v2.0.15):

> This release didn't include a release tarball. Furthermore, it had issues such as https://github.com/numactl/numactl/issues/135, https://github.com/numactl/numactl/issues/136 and https://github.com/numactl/numactl/issues/138 that got fixed in the follow up release. Please try to use v2.0.16 instead.

### CODE REVIEW

File: var/spack/repos/builtin/packages/numactl/package.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ class Numactl(AutotoolsPackage):
 
     license("LGPL-2.1-only")
 
+    version("2.0.18", sha256="8cd6c13f3096e9c2293c1d732f56e2aa37a7ada1a98deed3fac7bd6da1aaaaf6")
+    version("2.0.17", sha256="af22829cda8b5bdee3d280e61291697bbd3f9bd372afdf119c9348b88369d40b")
+    version("2.0.16", sha256="a35c3bdb3efab5c65927e0de5703227760b1101f5e27ab741d8f32b3d5f0a44c")
     version("2.0.14", sha256="1ee27abd07ff6ba140aaf9bc6379b37825e54496e01d6f7343330cf1a4487035")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/numactl/package.py:22]: There are a number of patches that apply to 2.0.14, including on pgi and nvhpc compilers. Did you make sure these are not needed anymore? They might be the cause of the failing build in CI.

Comment (dc-snl) [reply var/spack/repos/builtin/packages/numactl/package.py:22]: Ah, looks I missed few items for 2.0.16:

We still need [fix-empty-block.patch](https://github.com/spack/spack/blob/develop/var/spack/repos/builtin/packages/numactl/fix-empty-block.patch) in [2.0.16](https://github.com/numactl/numactl/blob/v2.0.16/configure.ac#L20), but not [2.0.17](https://github.com/numactl/numactl/blob/v2.0.17/configure.ac#L23).

We also need the [configure.ac](https://github.com/numactl/numactl/blob/v2.0.16/configure.ac) change from [link-with-latomic-if-needed.patch](https://github.com/spack/spack/blob/9b077a360e73ddeadf473f99530c082d9cb83685/var/spack/repos/builtin/packages/numactl/link-with-latomic-if-needed.patch) to link with libatomic in [2.0.16](https://github.com/numactl/numactl/blob/v2.0.16/configure.ac), but not [2.0.17](https://github.com/numactl/numactl/blob/v2.0.17/configure.ac#L30).

I'll update the package.py and pull the libatomic link into another patch by itself for 2.0.16.

**Comment by dc-snl** [comment]
As for the build error for 2.0.18, https://gitlab.spack.io/spack/spack/-/jobs/12383516#L359, I'll see if we need another patch or just some extra flags.

**Comment by dc-snl** [comment]
Looks like v2.0.18 required a new patch. I think this is now good to go, assuming CI passes.

File: var/spack/repos/builtin/packages/numactl/package.py [hunk 2]
```diff
@@ -25,8 +28,9 @@ class Numactl(AutotoolsPackage):
     patch("numactl-2.0.11-sysmacros.patch", when="@2.0.11")
     # https://github.com/numactl/numactl/issues/94
     patch("numactl-2.0.14-symver.patch", when="@2.0.14")
-    patch("fix-empty-block.patch", when="@2.0.10:2.0.14")
+    patch("fix-empty-block.patch", when="@2.0.10:2.0.16")
     patch("link-with-latomic-if-needed.patch", when="@2.0.14")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/numactl/package.py:32]: Based on https://github.com/spack/spack/pull/46150/files#r1742322060 I think this is the patch sequence then?
```suggestion
    patch("link-with-latomic-if-needed.patch", when="@2.0.14")
    patch("link-with-latomic-if-needed-v2.0.16.patch", when="@2.0.16")
```
Skipping 2.0.15 here too per summary.

Comment (dc-snl) [reply var/spack/repos/builtin/packages/numactl/package.py:32]: Oops. Yep. I forgot to add that patch to `package.py`. Thanks!

**Review by wdconinc — APPROVED** [review]
Thaniks. This looks good to me, and hopefully passes CI builds too :-)

**Comment by wdconinc** [comment]
FYI https://github.com/spack/spack-infrastructure/blob/main/docs/deferred_pipelines.md#tips

**Pull request merged**